distance-metrics/src/legacy.rs

```diff
@@ -0,0 +1,55 @@
+const DIMENSIONS: usize = 300;
+
+#[repr(align(32))]
+pub struct FloatArray(pub [f32; DIMENSIONS]);
+
```

Comment:
Style nit: so far, the crate uses a top-down convention where basic definitions like this are on the bottom rather than on top, so would be nice to move this down (and invert the order).